### PR TITLE
Add org.freac.freac

### PR DESCRIPTION
**Pull request by enzo1982**
This is the fre:ac audio converter. It should work on all supported architectures.

I followed the app submission and app requirements guidelines. Let me know if I can improve anything.

### CODE REVIEW

**Comment by barthalion** [comment]
bot, build org.freac.freac
testbot, build org.freac.freac

File: org.freac.freac.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.appdata.xml]: It's there already, but added after the current release.

https://github.com/enzo1982/freac/tree/master/metadata

File: org.freac.freac.desktop [hunk 2]
```diff
@@ -0,0 +1,8 @@
+[Desktop Entry]
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.desktop]: See appdata comment above.

File: org.freac.freac.json [hunk 3]
```diff
@@ -0,0 +1,376 @@
+{
+    "app-id": "org.freac.freac",
+    "runtime": "org.freedesktop.Platform",
+    "runtime-version": "18.08",
+    "sdk": "org.freedesktop.Sdk",
+    "command": "freac",
+    "finish-args": [
+        "--device=all",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: For CD ripping. Afaik there's no other way yet to access CD drives?

File: org.freac.freac.json [hunk 4]
```diff
@@ -0,0 +1,376 @@
...
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--filesystem=host",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--share=ipc",
+        "--share=network",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: For freedb/CDDB.

File: org.freac.freac.json [hunk 5]
```diff
@@ -0,0 +1,376 @@
...
+        "/share/aclocal",
+        "/share/info",
+        "/share/man",
+        "*.a",
+        "*.la"
+    ],
+    "modules": [
+        {
+            "name": "libcdio",
+            "buildsystem": "autotools",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: OK, will remove those lines.

File: org.freac.freac.json [hunk 6]
```diff
@@ -0,0 +1,376 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://github.com/uriparser/uriparser/releases/download/uriparser-0.9.0/uriparser-0.9.0.tar.bz2",
+                    "sha256": "ec67eb34feda8eac166f281799f03ed48387694fca44f6f5852f61f8fb535e2c"
+                }
+            ]
+        },
+        {
+            "name": "libvorbis",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: This one has the aoTuV tuning and Lancer intrinsics patches though.
https://wiki.hydrogenaud.io/index.php?title=AoTuV
https://wiki.hydrogenaud.io/index.php?title=Lancer

File: org.freac.freac.json [hunk 7]
```diff
@@ -0,0 +1,376 @@
...
+                {
+                    "type": "file",
+                    "url": "http://git.savannah.gnu.org/gitweb/?p=config.git;a=blob_plain;f=config.sub;hb=058639b",
+                    "sha256": "9afeeb8bf78f3ff549eb40f1364b54b4404ea1be27eee446af02790d9e39cc92",
+                    "dest-filename": "autoaux/config.sub"
+                }
+            ]
+        },
+        {
+            "name": "libogg",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: This one has the fastcrc patch applied, which has been upstreamed already, but is not available in a release yet.
https://git.xiph.org/?p=ogg.git;a=commit;h=bc82844df068429d209e909da47b1f730b53b689

File: org.freac.freac.json [hunk 8]
```diff
@@ -0,0 +1,376 @@
...
+            ]
+        },
+        {
+            "name": "freac",
+            "buildsystem": "simple",
+            "build-options": {
+                "cxxflags": "-DFREAC_DATE=\\\"20181201a\\\""
+            },
+            "build-commands": [
+                "make -j8 prefix=/app install"
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: Thanks, will change that. I saw other build recipes using -j8, didn't know about FLATPAK_BUILDER_N_JOBS.

File: org.freac.freac.json [hunk 9]
```diff
@@ -0,0 +1,376 @@
...
+            ],
+            "post-install": [
+                "install -d /app/share/applications /app/share/metainfo",
+                "install -d /app/share/icons/hicolor/64x64/apps /app/share/icons/hicolor/128x128/apps",
+                "install org.freac.freac.desktop /app/share/applications",
+                "install org.freac.freac.appdata.xml /app/share/metainfo",
+                "install org.freac.freac-64x64.png /app/share/icons/hicolor/64x64/apps/org.freac.freac.png",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: Like the appdata and desktop files, this has already been added after the current release.
https://github.com/enzo1982/freac/tree/master/icons

File: org.freac.freac.json [hunk 10]
```diff
@@ -0,0 +1,376 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://ftp.gnu.org/gnu/libcdio/libcdio-paranoia-10.2+0.94+2.tar.gz",
+                    "sha256": "d60f82ece97eeb92407a9ee03f3499c8983206672c28ae5e4e22179063c81941"
+                }
+            ]
+        },
+        {
+            "name": "libflac",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: Similar to libogg, this one has the fastcrc patch applied which has been upstreamed already.
https://git.xiph.org/?p=flac.git;a=commit;h=5579f2922f048aa3bd2b206dd64f6ead2846c8ac

**Comment by enzo1982** [comment]
Thanks for reviewing!

Updated to address the `"buildsystem": "autotools"` and `make -j8` remarks.

**Comment by barthalion** [comment]
bot, build org.freac.freac

**Comment by barthalion** [comment]
i386 failed: https://flathub.org/builds/#/builders/23/builds/704

(testbot checks only x86_64 for now)

File: org.freac.freac.appdata.xml [hunk 11]
```diff
@@ -0,0 +1,82 @@
...
+        </screenshot>
+        <screenshot>
+            <image>https://www.freac.org/images/screenshots/freac-linux-tageditor.png</image>
+        </screenshot>
+        <screenshot>
+            <image>https://www.freac.org/images/screenshots/freac-linux-settings.png</image>
+        </screenshot>
+    </screenshots>
+    <releases>
+        <release version="1.1 alpha 20181201a" date="2018-12-02">
```
Comment (TingPing) [inline org.freac.freac.appdata.xml]: You might encounter problems with this versioning later, they are used to compare versions.

**Comment by nedrichards** [comment]
bot, build org.freac.freac
testbot, build org.freac.freac

File: org.freac.freac.appdata.xml (hunk 11 above)
Comment (enzo1982) [reply org.freac.freac.appdata.xml]: OK, I'll use `1.1~alpha20181201a`. That nicely sorts before even `1.1` and is what is already used on e.g. Haiku.

File: org.freac.freac.json [hunk 12]
```diff
@@ -0,0 +1,366 @@
...
+    "finish-args": [
+        "--device=all",
+        "--env=DCONF_USER_CONFIG_DIR=.config/dconf",
+        "--filesystem=host",
+        "--filesystem=xdg-run/dconf",
+        "--filesystem=~/.config/dconf:ro",
+        "--share=ipc",
+        "--share=network",
+        "--socket=pulseaudio",
+        "--socket=x11",
```
[the remark this answers is not in the document]
Comment (enzo1982) [reply org.freac.freac.json]: No, fre:ac uses my smooth Class Library which does not support Wayland yet.

**Comment by enzo1982** [comment]
Updated again to address the remarks from @TingPing.

**Comment by barthalion** [comment]
bot, build org.freac.freac

**Comment by enzo1982** [comment]
Looks like the x86_64 build failed due to a build system issue.

**Comment by barthalion** [comment]
Repository has been created: https://github.com/flathub/org.freac.freac

Thanks!

**Pull request closed without merging**